src/services/Middleware.ts

```diff
@@ -0,0 +1,56 @@
+import * as Hapi from 'hapi'
+import * as Boom from 'boom'
+import { create as createLogger } from '../common/log'
+import Config from '../services/Config'
+import SelfTest from '../services/SelfTest'
+import { Injector } from 'reduct'
+const log = createLogger('Middleware')
+
+export default class Middleware {
+  public checkSelfTestPlugin = {
+    name: 'Self Test Checker',
+    version: '1.0.0',
+    register: async (server: Hapi.Server, options: any) => {
+      server.ext('onPreAuth', (request: Hapi.Request, h: Hapi.ResponseToolkit) => {
+        const config = this.config
+        const selfTest = this.selfTest
+
+        if (!config) {
```

Comment:
In what scenario could this end up unspecified? I don't think we need to validate that the dependency injection worked